fluentnamer/src/main/java/com/azure/autorest/fluent/transformer/ResourceTypeNormalization.java

```diff
@@ -295,6 +291,35 @@ private static void adaptForParentSchema(ObjectSchema compositeType, ResourceTyp
 
             LOGGER.info("Change parent from '{}' to '{}', for '{}'", Utils.getJavaName(parentType), type.getClassName(), Utils.getJavaName(compositeType));
         }
+
+        if (type.getClassName().equals(Utils.getJavaName(compositeType))) {
```

Comment:
This is to handle that current model be `ProxyResource`, and its parent also be `ProxyResource`.

On ModelMapper, it would be regarded as "external model".

But when generating sample/test, the util will get into infinite loop on the parent class...

Reply:
One concern (which is not related to this change) is that what if this model is called `ProxyResource`, but having some property not in a standard `ProxyResource`.